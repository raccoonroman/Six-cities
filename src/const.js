export const DEFAULT_CITIES = [`Paris`, `Cologne`, `Brussels`, `Amsterdam`, `Hamburg`, `Dusseldorf`];

export const SortType = {
  POPULAR: `Popular`,
  PRICE_LOW_TO_HIGH: `Price: low to high`,
  PRICE_HIGH_TO_LOW: `Price: high to low`,
  TOP_RATED_FIRST: `Top rated first`,
};

export const CardType = {
  CITY: `city`,
  NEAR: `near`,
  FAVORITE: `favorite`,
};

export const ActionType = {
  LOAD_OFFERS: `LOAD_OFFERS`,
  UPDATE_OFFER: `UPDATE_OFFER`,
  SET_CITY: `SET_CITY`,
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
  SET_EMAIL: `SET_EMAIL`,
  LOAD_COMMENTS: `LOAD_COMMENTS`,
  LOAD_NEARBY_OFFERS: `LOAD_NEARBY_OFFERS`,
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const AppRoute = {
  ROOT: `/`,
  LOGIN: `/login`,
  FAVORITES: `/favorites`,
  OFFER: `/offer`,
};
