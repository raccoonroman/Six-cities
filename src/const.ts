export const DEFAULT_CITIES: string[] = ['Paris', 'Cologne', 'Brussels', 'Amsterdam', 'Hamburg', 'Dusseldorf'];

export enum SortType {
  POPULAR = 'Popular',
  PRICE_LOW_TO_HIGH = 'Price: low to high',
  PRICE_HIGH_TO_LOW = 'Price: high to low',
  TOP_RATED_FIRST = 'Top rated first',
}

export enum CardType {
  CITY = 'city',
  NEAR = 'near',
  FAVORITE = 'favorite',
}

export enum ActionType {
  LOAD_OFFERS = 'LOAD_OFFERS',
  UPDATE_OFFER = 'UPDATE_OFFER',
  SET_CITY = 'SET_CITY',
  REQUIRED_AUTHORIZATION = 'REQUIRED_AUTHORIZATION',
  SET_EMAIL = 'SET_EMAIL',
  LOAD_COMMENTS = 'LOAD_COMMENTS',
  LOAD_NEARBY_OFFERS = 'LOAD_NEARBY_OFFERS',
}

export enum AuthorizationStatus {
  AUTH = 'AUTH',
  NO_AUTH = 'NO_AUTH',
}

export enum AppRoute {
  ROOT = '/',
  LOGIN = '/login',
  FAVORITES = '/favorites',
  OFFER = '/offer',
}

export enum MapType {
  HOVERED_ACTIVE_OFFER = 'HOVERED_ACTIVE_OFFER',
  STATIC_ACTIVE_OFFER = 'STATIC_ACTIVE_OFFER',
}
