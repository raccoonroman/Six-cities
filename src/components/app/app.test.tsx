import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { AuthorizationStatus } from '@/const';
import App from '@/components/app';


const mockStore = configureStore([]);

const cities = ['Vinnytsia', 'Kyiv'];


it('Render <App />', () => {
  const store = mockStore({
    offers: {
      offers: [],
    },
    cities: {
      currentCity: cities[0],
      cities,
    },
    authorization: {
      authorization: AuthorizationStatus.NO_AUTH,
    },
    userData: {
      email: '',
    },
  });

  const tree = renderer.create(
    <Provider store={store}>
      <App />
    </Provider>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
