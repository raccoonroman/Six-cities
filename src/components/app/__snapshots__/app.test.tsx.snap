// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="/img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/login"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__login"
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index page__main--index-empty"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <ul
          className="locations__list tabs__list"
        >
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item tabs__item--active"
              href="#"
              onClick={[Function]}
            >
              <span>
                Vinnytsia
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item"
              href="#"
              onClick={[Function]}
            >
              <span>
                Kyiv
              </span>
            </a>
          </li>
        </ul>
      </section>
    </div>
    <div
      className="cities"
    >
      <div
        className="cities__places-container container cities__places-container--empty"
      >
        <section
          className="cities__no-places"
        >
          <div
            className="cities__status-wrapper tabs__content"
          >
            <b
              className="cities__status"
            >
              No places to stay available
            </b>
            <p
              className="cities__status-description"
            >
              We could not find any property available at the moment in 
              Vinnytsia
            </p>
          </div>
        </section>
        <div
          className="cities__right-section"
        />
      </div>
    </div>
  </main>
</div>
`;
